test/jasmine/tests/funnelarea_test.js

```diff
@@ -0,0 +1,1408 @@
+              .then(_checkFontSizes([24, 34, 12, 12, 12, 12]))
+              .catch(failTest)
+              .then(done);
+        });
+    });
+
+    it('should fall back to textfont array values and layout.font scalar (except color)' +
+      ' values for inside text', function(done) {
+        var layout = {font: {color: 'orange', family: 'serif', size: 16}};
+        var data = Lib.extendFlat({}, insideTextTestsTrace, {
+            textfont: {
+                color: ['blue', 'blue'], family: ['Arial', 'Arial'], size: [18, 18]
+            },
+            insidetextfont: {
+                color: ['purple'], family: ['Roboto'], size: [24]
+            }
+        });
+
+        Plotly.plot(gd, [data], layout)
+          .then(_checkFontColors(['purple', 'blue', LIGHT, LIGHT, DARK, LIGHT]))
+          .then(_checkFontFamilies(['Roboto', 'Arial', 'serif', 'serif', 'serif', 'serif']))
+          .then(_checkFontSizes([24, 18, 16, 16, 16, 16]))
+          .catch(failTest)
+          .then(done);
+    });
+
+    [
+        {fontAttr: 'textfont'},
+        {fontAttr: 'insidetextfont'}
+    ].forEach(function(spec) {
+        it('should fall back to layout.font scalar values for inside text (except color) if ' + spec.fontAttr + ' value ' +
+          'arrays don\'t cover all slices', function(done) {
+            var layout = {font: {color: 'orange', family: 'serif', size: 16}};
+            var data = Lib.extendFlat({}, insideTextTestsTrace);
+            data.textposition = 'inside';
+            data[spec.fontAttr] = {color: ['blue', 'yellow'], family: ['Arial', 'Arial'], size: [24, 34]};
+
+            Plotly.plot(gd, [data], layout)
+              .then(_checkFontColors(['blue', 'yellow', LIGHT, LIGHT, DARK, LIGHT]))
+              .then(_checkFontFamilies(['Arial', 'Arial', 'serif', 'serif', 'serif', 'serif']))
+              .then(_checkFontSizes([24, 34, 16, 16, 16, 16]))
+              .catch(failTest)
+              .then(done);
+        });
+    });
+
+    function _assertTitle(msg, expText, expColor) {
+        var title = d3.select('.titletext > text');
+        expect(title.text()).toBe(expText, msg + ' text');
+        expect(title.node().style.fill).toBe(expColor, msg + ' color');
+    }
+
+    it('show a user-defined title with a custom position and font', function(done) {
+        Plotly.plot(gd, [{
+            type: 'funnelarea',
+            values: [1, 2, 3],
+            title: {
+                text: 'yo',
+                font: {color: 'blue'},
+                position: 'top left'
+            }
+        }])
+          .then(function() {
+              _assertTitle('base', 'yo', 'rgb(0, 0, 255)');
+              _verifyTitle(true, false, true, false, false);
+          })
+          .catch(failTest)
+          .then(done);
+    });
+
+    it('still support the deprecated `title` structure (backwards-compatibility)', function(done) {
```

Comment:
Hmm. Maybe we shouldn't support `title: 'text'` signature for new trace types.

Reply:
Thinking about this again. I think we should go all-in with the new API. Let's **not** support `title: 'text'`.

Reply:
Good. & What about `funnel`?

Reply:
`funnel` doesn't have a `title` attribute.

Reply:
Good point. As another point the `funnelarea` mocks are using the new API. But we may need to drop/adapt few jasmine tests.